### DISCUSSION TITLE
Budibase REST API

### DISCUSSION THREAD

**Discussion by cguerin** [Budibase/budibase#3748] [Help] [answered]
Is there anyway I can put a REST API on Budibase to query it's internal databases for tables I create in it, from remote?

**Comment by mjashanks** [comment, marked as answer]
Budibase has a REST api with all the endpoints that you need for this. Unfortunately, we have not documented it yet - or provided a great auth mechanism.

If you are self-hosting, you can follow these steps to use our API: https://github.com/Budibase/budibase/discussions/1915#discussioncomment-966149 (except step 1).

You might have open an app and watch your `dev-tools > network tab`, to see the various calls that happen.

**Reply by pigeonflight** [reply]
#1915 refers to public access 
![image](https://user-images.githubusercontent.com/31827/150514940-ba5c2306-3778-4014-830e-83d5d31df4b7.png)

What about endpoints for "non-public" tables?

**Comment by cguerin** [comment]
Oh super, I’ll go play with that.
That’s exactly what I needed.
